src/test/java/seedu/tp/utils/InputTestUtil.java

```diff
@@ -22,4 +26,28 @@ public static FlashcardFactory getFlashcardFactoryWithInput(String simulatedInpu
         System.setIn(SYS_IN_BACKUP);
         return flashcardFactory;
     }
+
+    public static GroupFactory getGroupFactoryWithInput(String simulatedInput, FlashcardList flashcardList) {
+        ByteArrayInputStream simulatedSystemIn = new ByteArrayInputStream(simulatedInput.getBytes());
+        System.setIn(simulatedSystemIn);
+        GroupFactory groupFactory = new GroupFactory(new Ui(),flashcardList);
+        System.setIn(SYS_IN_BACKUP);
+        return groupFactory;
+    }
+
+    public static AddFlashcardToGroupCommand getAddFlashcardToGroupCommandWithInput
+            (String simulatedInput, FlashcardList flashcardList, GroupList originalGroupList) {
+        ByteArrayInputStream simulatedSystemIn = new ByteArrayInputStream(simulatedInput.getBytes());
+        System.setIn(simulatedSystemIn);
+        return new AddFlashcardToGroupCommand(new Ui(), originalGroupList, flashcardList);
+    }
+
+    public static int[] interpretIndexes(String givenIndexes) {
```

Comment:
Think this can be given a more descriptive name
i.e. convertStringIndexesToIntArray
or something like that